server/store/pluginstore/store.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package pluginstore
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+type User struct {
+	MattermostUserID string
+	TeamsObjectID    string
+	TeamsSSOUsername string
+}
+
+func NewUser(mattermostUserID, teamsObjectID, teamsSSOUsername string) *User {
+	return &User{
+		MattermostUserID: mattermostUserID,
+		TeamsObjectID:    teamsObjectID,
+		TeamsSSOUsername: teamsSSOUsername,
+	}
+}
+
+type Store interface {
```

Comment:
Do we need a `Store` for this? The plugin API wrapper already has a `Get(key string, o any)` interface that does the JSON unmarshalling (similar for `Set`).

Reply:
I made it a separate file so it was easier to move to the devsecops application, the other store seemed more focused into direct SQL storage.

Reply:
Yeah, totally fine to keep them separate, I mean just use the pluginapi directly without any abstraction layer, but 0/5 given this is moving.